component/cortex-tenant-ns-label.jsonnet

```diff
@@ -0,0 +1,125 @@
+local kap = import 'lib/kapitan.libjsonnet';
+local kube = import 'lib/kube.libjsonnet';
+
+local inv = kap.inventory();
+local params = inv.parameters.cortex_tenant_ns_label;
+
+local namespace_meta = {
+  metadata+: {
+    namespace: params.namespace,
+  },
+};
+
+local namespace = kube.Namespace(params.namespace);
+
+local secret = kube.Secret(params.app_name) + namespace_meta {
+  stringData: {
+    CONFIG: params.config,
+  },
+};
+
+
+local deployment = kube.Deployment('cortex-tenant-ns-label') + namespace_meta {
+  spec+: {
+    template+: {
+      metadata+: {
+        labels+: {
+          app: 'cortex-tenant-ns-label',
+        },
+      },
+      spec+: {
+        containers_:: {
+          [params.app_name]: kube.Container(params.app_name) {
+            image: '%(registry)s/%(repository)s:%(tag)s' % params.image,
+            resources: {
+              limits: {
+                cpu: params.limits.cpu,
+                memory: params.limits.memory,
+              },
+              requests: {
+                cpu: params.requests.cpu,
+                memory: params.requests.memory,
+              },
+            },
+            ports_:: {
+              http: { containerPort: 8080 },
+            },
+            livenessProbe: {
+              httpGet: {
+                path: '/alive',
+                port: 8080,
+              },
+              initialDelaySeconds: 30,
+              periodSeconds: 30,
+            },
+            envFrom: [
+              {
+                secretRef: {
+                  name: secret.metadata.name,
+                },
+              },
+            ],
+          },
+        },
+        serviceAccountName+: params.app_name,
+      },
+    },
+  },
+};
+
+local service = kube.Service('cortex-tenant-ns-label') + namespace_meta {
+  target_pod:: deployment.spec.template,
+  target_container_name:: params.app_name,
+};
+
+local service_account = kube.ServiceAccount(params.app_name) + namespace_meta {
+};
+
+local cluster_role = kube.ClusterRole(params.app_name + ':namespace-reader') {
+  rules: [
+    { apiGroups: [ '' ], resources: [ 'namespaces' ], verbs: [ 'get', 'list' ] },
+  ],
+};
+
+local cluster_role_binding = kube.ClusterRoleBinding(params.app_name + '-namespace-reader') {
+  subjects_: [ service_account ],
+  roleRef_: cluster_role,
+};
+
+local network_policy = kube.NetworkPolicy('allow-from-prometheus') + namespace_meta {
```

Comment:
This network policy only makes sense for OpenShift 4, since other setups will have the Prometheus deployments in different namespaces, possibly with different names. However, it's probably fine to keep it as is for now.